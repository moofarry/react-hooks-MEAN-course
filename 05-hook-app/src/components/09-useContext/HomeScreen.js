import React from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {

  const { user } = React.useContext(UserContext);

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />

      <pre>
        {JSON.stringify( user,null,3)}
      </pre>
    </div>
  );
};
