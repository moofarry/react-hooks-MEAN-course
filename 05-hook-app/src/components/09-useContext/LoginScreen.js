import React from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const { setUser } = React.useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />

      <button className="btn btn-primary" onClick={() => setUser({ id: 123, name: "jhon" })}>
        Login
      </button>
    </div>
  );
};
