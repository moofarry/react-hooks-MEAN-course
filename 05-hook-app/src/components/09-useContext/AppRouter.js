import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <NavBar />
      <div className="container">
        <Routes>
          <Route path="/" element={<HomeScreen />} />
          <Route path="/about" element={<AboutScreen />} />
          <Route path="/login" element={<LoginScreen />} />

          <Route path="*" element={<HomeScreen />} />
        </Routes>
      </div>
    </Router>
  );
};
