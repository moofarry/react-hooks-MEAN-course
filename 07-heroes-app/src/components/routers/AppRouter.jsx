import { BrowserRouter, Route, Routes } from "react-router-dom";
import { LoginScreen } from "../login/LoginScreen";
import { DashBoardRoutes } from "./DashBoardRoutes";

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/login" element={<LoginScreen />} />
        <Route path="/*" element={<DashBoardRoutes />} />
      </Routes>
    </BrowserRouter>
  );
};
