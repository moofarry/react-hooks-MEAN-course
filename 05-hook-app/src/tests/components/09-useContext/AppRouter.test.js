import React from "react";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { mount } from "enzyme";

import { UserContext } from "../../../components/09-useContext/UserContext";

describe("test in <AppRouter/>", () => {
  const user = {
    name: "jhon",
    email: "jhon@yopmail.com",
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );

  test("should be show correctly", () => {

    expect(wrapper).toMatchSnapshot();
  });
});
