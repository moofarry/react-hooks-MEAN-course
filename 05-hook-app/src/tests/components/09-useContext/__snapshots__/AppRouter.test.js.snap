// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <AppRouter/> should be show correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
       useContext 
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link navData => navData.isActive && \\"active\\" active"
          href="/"
          onClick={[Function]}
        >
           Home 
        </a>
        <a
          className="nav-link navData => navData.isActive && \\"active\\""
          href="/about"
          onClick={[Function]}
        >
           About 
        </a>
        <a
          className="nav-link navData => navData.isActive && \\"active\\""
          href="/login"
          onClick={[Function]}
        >
           Login
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "jhon",
   "email": "jhon@yopmail.com"
}
      </pre>
    </div>
  </div>,
]
`;
